src/main/java/com/example/demo/services/DefaultPushNotificationsService.java

```diff
@@ -0,0 +1,191 @@
+package com.example.demo.services;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpRequest.Builder;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.ECPrivateKey;
+import java.security.interfaces.ECPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.sql.Timestamp;
+import java.util.Base64;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.springframework.data.domain.PageRequest;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.example.demo.entities.Notification;
+import com.example.demo.entities.PushSub;
+import com.example.demo.entities.User;
+import com.example.demo.exceptions.AuctionAppException;
+import com.example.demo.exceptions.BadInitializatinException;
+import com.example.demo.exceptions.InvalidDataException;
+import com.example.demo.exceptions.UnallowedOperationException;
+import com.example.demo.models.NotificationModel;
+import com.example.demo.models.PushServiceSubModel;
+import com.example.demo.repositories.NotificationsRepository;
+import com.example.demo.repositories.PushSubsRepository;
+import com.example.demo.services.interfaces.PushNotificationsService;
+import com.example.demo.utils.PushMessageEncryptionUtil;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+
+public class DefaultPushNotificationsService implements PushNotificationsService{
+	
+	private static final long HALF_DAY_MILIS=12 * 60 * 60 * 1000;
+	private static final byte[] P256_HEAD= Base64.getDecoder().decode("MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgA");
```

Comment:
what is this secret key, think it should be in secrets and passed with environmental variables

Reply:
This is like some header for the ec p256 algorithm, and it is only used for its length. When extracting a raw key from some encoded byte array you have to discard the first n bytes, where n is the length of header